pkg/oc/cli/admin/migrate/scc/scc.go

```diff
@@ -0,0 +1,392 @@
+
+var (
+	internalMigrateSCCShort = "Converts SCCs to similar PSPs"
+	internalMigrateSCCLong  = templates.LongDesc(`
+		Converts SecurityContextConstraints to the equivalent PodSecurityPolicy objects.
+		For each SCC it also creates a ClusterRole with the similar name and ClusterRoleBinding
+		for every user or group that SCC has. To easy identify such auto-generated roles and bindings,
+		their names are prefixed by 'psp:' string.
+
+		NOTE: it does not modify anything by default.`)
+
+	internalMigrateSCCExample = templates.Examples(`
+		# Perform a dry-run of migrating all objects
+		%[1]s
+
+		# To actually perform the migration, the confirm flag must be appended
+		%[1]s --confirm`)
+)
+
+type migrateSCCOptions struct {
+	kubeRbacClient   *rbacv1client.RbacV1Client
+	kubePolicyClient *policyv1beta1client.PolicyV1beta1Client
+
+	cachedRoles        *rbacv1.RoleList
+	cachedClusterRoles *rbacv1.ClusterRoleList
+
+	migrate.ResourceOptions
+}
+
+func newCmdMigrateSCCOptions(out, errout io.Writer) *migrateSCCOptions {
+	return &migrateSCCOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			Out:    out,
+			ErrOut: errout,
+
+			AllNamespaces: true,
+			Include:       []string{"securitycontextconstraints.security.openshift.io"},
+		},
+	}
+}
+
+func NewCmdMigrateSCC(name, fullName string, f kcmdutil.Factory, _ io.Reader, out, errout io.Writer) *cobra.Command {
+	options := newCmdMigrateSCCOptions(out, errout)
+
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   internalMigrateSCCShort,
+		Long:    internalMigrateSCCLong,
+		Example: fmt.Sprintf(internalMigrateSCCExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(name, f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run())
+		},
+	}
+	options.ResourceOptions.Bind(cmd)
+
+	// these flags aren't applied for SCC migrations
+	cmd.Flags().MarkHidden("to-key")
+	cmd.Flags().MarkHidden("from-key")
+	cmd.Flags().MarkHidden("include")
+	cmd.Flags().MarkHidden("filename")
+	cmd.Flags().MarkHidden("show-all")
+	cmd.Flags().MarkHidden("no-headers")
+	cmd.Flags().MarkHidden("show-labels")
+	cmd.Flags().MarkHidden("all-namespaces")
+
+	return cmd
+}
+
+func (o *migrateSCCOptions) Complete(name string, f kcmdutil.Factory, c *cobra.Command, args []string) error {
+	o.ResourceOptions.SaveFn = o.save
+	if err := o.ResourceOptions.Complete(f, c); err != nil {
+		return err
+	}
+	clientConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	kubeRbacClient, err := rbacv1client.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	o.kubeRbacClient = kubeRbacClient
+
+	kubePolicyClient, err := policyv1beta1client.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	o.kubePolicyClient = kubePolicyClient
+
+	o.cachedClusterRoles, err = o.kubeRbacClient.ClusterRoles().List(metav1.ListOptions{})
```

Comment:
This belongs in `Run`, complete is only to fill in `migrateSCCOptions`. Only `Run` should contain any actions.